python/src/scipp/io/hdf5.py

```diff
@@ -0,0 +1,102 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Simon Heybrock
+
+
+def _unit_lut():
+    from .._scipp.core import units as u
+    units = [u.one, u.m, u.kg, u.us, u.angstrom]
+    names = [str(unit) for unit in units]
+    return dict(zip(names, units))
+
+
+class HDF5IO:
+    _units = _unit_lut()
+
+    def _read_unit(self, dset):
+        return self._units[dset.attrs['unit']]
+
+    def _write_variable(self, group, var, name):
+        from .._scipp import core as sc
+        if var.dtype in [sc.dtype.string, sc.dtype.PyObject]:
+            print(f'Writing with dtype={var.dtype} not implemented, skipping.')
+            return
+        group = group.create_group(name)
+        dset = group.create_dataset('values', data=var.values)
+        dset.attrs['dims'] = [str(dim) for dim in var.dims]
+        dset.attrs['dtype'] = str(var.dtype)
+        dset.attrs['unit'] = str(var.unit)
+        if var.variances is not None:
+            variances = group.create_dataset('variances', data=var.variances)
+            dset.attrs['variances'] = variances.ref
+
+    def _read_variable(self, group):
+        from .._scipp import core as sc
+        import numpy as np
+        values = group['values']
+        dtype = values.attrs['dtype']
+        contents = dict()
+        # For some reason we cannot pass the dataset directly to scipp, and
+        # we also need to manually override the dtype. Is h5py using a
+        # different float type?
```

Comment:
@OwenArnold @nvaytet Any ideas why this is not working? I had assumed/hoped that the `py::array` interface would just work here.

Reply:
Not clear to me why this isn't working. What error do you get?

Reply:
The usual error from pybind11 when passing a type with no matching overload found.